src/compile/scale/type.ts

```diff
@@ -77,7 +77,7 @@ function defaultType(
       }
 
       if (isXorY(channel) || isXorYOffset(channel)) {
-        if (util.contains(['rect', 'bar', 'image', 'rule'], mark.type)) {
+        if (util.contains(['rect', 'bar', 'image', 'tick', 'rule'], mark.type)) {
```

Comment:
Need to make tick use band scale by default, just like bar/rect, to leverage the same set of logic.